packages/iov-bns/src/integrationtests/multisignature.spec.ts

```diff
@@ -0,0 +1,286 @@
+import {
+  Address,
+  Amount,
+  ChainId,
+  Identity,
+  isBlockInfoPending,
+  isBlockInfoSucceeded,
+  SendTransaction,
+  SignedTransaction,
+  TokenTicker,
+  UnsignedTransaction,
+  WithCreator,
+} from "@iov/bcp";
+import { Slip10RawIndex } from "@iov/crypto";
+import { Ed25519HdWallet, HdPaths, UserProfile } from "@iov/keycontrol";
+import BN from "bn.js";
+
+import { bnsCodec } from "../bnscodec";
+import { BnsConnection } from "../bnsconnection";
+import { CreateMultisignatureTx, MultisigTransaction, Participant } from "../types";
+import { conditionToAddress, multisignatureCondition } from "../util";
+
+const CASH = "CASH" as TokenTicker;
+const bnsUrl = "ws://localhost:23456";
+const chainId = "local-iov-devnet" as ChainId;
+// Address: tiov1xwvnaxahzcszkvmk362m7vndjkzumv8ufmzy3m
+const aliceMnemonic = "host century wave huge seed boost success right brave general orphan lion";
+// Address: tiov1qrw95py2x7fzjw25euuqlj6dq6t0jahe7rh8wp
+const bobMnemonic = "dad kiss slogan offer outer bomb usual dream awkward jeans enlist mansion";
+const defaultFeeAmount = 10000000;
+
+async function sleep(ms: number): Promise<void> {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+async function tendermintSearchIndexUpdated(): Promise<void> {
+  // Tendermint needs some time before a committed transaction is found in search
+  return sleep(50);
+}
+
+function pendingWithoutBnsd(): void {
+  if (!process.env.BNSD_ENABLED) {
+    pending("Set BNSD_ENABLED to enable bnsd-based tests");
+  }
+}
+
+interface ActorData {
+  readonly profile: UserProfile;
+  readonly connection: BnsConnection;
+  readonly identity: Identity;
+}
+
+class Actor {
+  public static async create(mnemonic: string, hdPath: readonly Slip10RawIndex[]): Promise<Actor> {
+    const profile = new UserProfile();
+    const wallet = profile.addWallet(Ed25519HdWallet.fromMnemonic(mnemonic));
+
+    const connection = await BnsConnection.establish(bnsUrl);
+
+    const identity = await profile.createIdentity(wallet.id, connection.chainId(), hdPath);
+
+    return new Actor({
+      profile: profile,
+      connection: connection,
+      identity: identity,
+    });
+  }
+
+  public get address(): Address {
+    return bnsCodec.identityToAddress(this.identity);
+  }
+
+  private readonly connection: BnsConnection;
+  private readonly profile: UserProfile;
+  private readonly identity: Identity;
+
+  public constructor(data: ActorData) {
+    this.profile = data.profile;
+    this.connection = data.connection;
+    this.identity = data.identity;
+  }
+
+  public async getBalance(): Promise<BN> {
+    const account = await this.connection.getAccount({ pubkey: this.identity.pubkey });
+    const balance = account ? account.balance : [];
+    const amount = balance.find(row => row.tokenTicker === CASH);
+    return new BN(amount ? amount.quantity : 0);
+  }
+
+  public async getMultisignatureBalance(address: Address): Promise<BN> {
```

Comment:
This works for any address, right? Why not merge into

```ts
/** get balance from address. Defaults to currect actor's address */
public async getBalance(address?: Address): Promise<BN> {
```

Reply:
Better yet, we don't actually use the current `getBalance`, so I'll remove it and rename this one.